bpf/process/types/basic.h

```diff
@@ -2378,6 +2383,12 @@ read_call_arg(void *ctx, struct msg_generic_kprobe *e, int index, int type,
 	case iov_iter_type:
 		size = copy_iov_iter(ctx, orig_off, arg, argm, e, data_heap);
 		break;
+	case linux_binprm_type: {
+		struct linux_binprm *bprm = (struct linux_binprm *)arg;
+		arg = (unsigned long)(&bprm->file);
```

Comment:
I believe that you should use CORE here as the linux_binprm can have different layout in different kernels. Something like `arg = (unsigned long)_(&bprm->file);` in a similar way to the line 2396 would be fine.

Reply:
Thanks!

I've never seen `_` used this way! Is this something particular to Tetragon? I expected to see `BPF_CORE_READ*` et al for CO:RE ops =)

Reply:
This is a macro you can find in `bpf/lib/bpf_helpers.h`:
```C
#define _(P) (__builtin_preserve_access_index(P))
```